src/com/google/javascript/jscomp/GlobalNamespace.java

```diff
@@ -1075,9 +1088,22 @@ boolean isSimpleStubDeclaration() {
     }
 
     boolean canCollapse() {
-      return !inExterns && !isGetOrSetDefinition() && (declaredType ||
+      boolean isAnnotatedNoCollapse = docInfo == null ?
+          false : docInfo.isNoCollapse();
+      if (!isAnnotatedNoCollapse) {
+        Ref ref = this.getDeclaration();
+        if (ref != null && ref.node != null) {
+          JSDocInfo info = NodeUtil.getBestJSDocInfo(ref.node);
```

Comment:
For an object literal, the docInfo points at the root object. Since a `@nocollapse` annotation could be used on a property key, we need to check that node as well. See test https://github.com/google/closure-compiler/pull/805/files#diff-f6ebc91c417a43a65475b9a83521bc54R84

I've added a comment to clarify.